packages/design-system/src/components/Drawer/DrawerManager.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Button } from '../Button';
+import { Drawer } from './Drawer';
+import { DrawerManager, useDrawerManager } from './DrawerManager';
+
+const SingleDrawer = ({ heading }) => {
+  const { toggleClick, closeClick, isOpen } = useDrawerManager();
+
+  return (
+    <>
+      {isOpen && (
+        <Drawer onCloseClick={closeClick} heading={heading}>
+          A drawer for {heading}
+        </Drawer>
+      )}
+      <Button className="ds-c-drawer__toggle" variation="ghost" onClick={toggleClick}>
+        toggle {heading}
+      </Button>
+    </>
+  );
+};
+
+const renderDrawerManager = () => {
+  return render(
+    <DrawerManager>
+      <p>Dummy content</p>
+      <div>
+        <SingleDrawer heading="drawer one" />
+        <SingleDrawer heading="drawer two" />
+        <SingleDrawer heading="drawer three" />
+      </div>
+    </DrawerManager>
+  );
+};
+
+describe('DrawerManager', () => {
+  it('renders a dialog', () => {
+    renderDrawerManager();
+    userEvent.click(screen.getAllByRole('button')[0]);
+    expect(screen.getByRole('dialog')).toMatchSnapshot();
```

Comment:
I actually don't think this snapshot is super necessary as long as we can `expect(screen.getByRole('dialog')).toBeInTheDocument()`. We're not so much testing the rendering of the drawer internals.